src/attributes/debugger.md

```diff
@@ -0,0 +1,158 @@
+Currently, this attribute only supports embedding Natvis files on `-windows-msvc` targets.
+</div>
+
+Consider a crate with this directory structure:
+
+```text
+/Cargo.toml
+/Rectangle.natvis
+  +-- src
+      +-- main.rs
+```
+
+Where `main.rs` contains:
+
+```rust ignore
+#![debugger_visualizer(natvis_file = "../Rectangle.natvis")]
+struct FancyRect {
+    pub x: f32,
+    pub y: f32,
+    pub dx: f32,
+    pub dy: f32,
+}
+
+fn main() {
+    let fancy_rect = FancyRect::new(10.0, 10.0, 5.0, 5.0);
+    Ok(());
+}
+```
+
+and `Rectangle.natvis` contains:
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<AutoVisualizer xmlns="http://schemas.microsoft.com/vstudio/debugger/natvis/2010">
+    <Type Name="Rectangle::FancyRect">
+      <DisplayString>({x},{y}) + ({dx}, {dy})</DisplayString>
+      <Expand>
+        <Synthetic Name="LowerLeft">
+          <DisplayString>({x}, {y})</DisplayString>
+        </Synthetic>
+        <Synthetic Name="UpperLeft">
+          <DisplayString>({x}, {y + dy})</DisplayString>
+        </Synthetic>
+        <Synthetic Name="UpperRight">
+          <DisplayString>({x + dx}, {y + dy})</DisplayString>
+        </Synthetic>
+        <Synthetic Name="LowerRight">
+          <DisplayString>({x + dx}, {y})</DisplayString>
+        </Synthetic>
+      </Expand>
+    </Type>
+</AutoVisualizer>
+```
+
+When viewed under WinDbg, the `fancy_rect` variable would be shown as follows:
+
+```text
+> Variables:
+  > fancy_rect: (10, 10) + (5, 5)
+    > LowerLeft: (10, 10)
+    > UpperLeft: (10, 15)
+    > UpperRight: (15, 15)
+    > LowerRight: (15, 10)
+```
+
+### Using `debugger_visualizer` with GDB
+
+GDB supports the use of a structured Python script, called a *pretty printer*, that describes how a type should be visualized in the debugger view.
+These scripts are embedded using the `gdb_script_file` meta item.
+For detailed information on pretty printers, refer to GDB's [pretty print documentation].
+
+Embedded pretty printers are not automatically loaded when debugging a binary under GDB.
+There are two ways to enable auto-loading embedded pretty printers:
+1. Launch GDB with extra arguments to explicitly add a directory or binary to the auto-load safe path: `gdb -iex "set auto-load safe-path path/to/binary" path/to/binary` (For more information, see GDB's [auto-loading documentation])
+1. Create a file named `gdbinit` under `$HOME/.config/gdb` (you may need to create the directory if it doesn't already exist). Add the following line to that file: `add-auto-load-safe-path path/to/binary`.
+
+Consider a crate called `PersonPrinter` with this directory structure:
+
+```text
+/Cargo.toml
+/printer.py
+  +-- src
+      +-- main.rs
+```
+
+Where `main.rs` contains:
+
+```rust ignore
+#![debugger_visualizer(gdb_script_file = "../printer.py")]
+mod person {
+    pub struct Person {
+        pub name: String,
+        pub age: i32,
+    }
+}
+
+use person::Person;
+
+fn main() {
+    let bob = Person::new(String::from("Bob"), 10);
```

Comment:
This example doesn't compile, which is a bit annoying when trying out the feature

Reply:
Good point - I just resolved the compile errors in both code snippets.